packages/react-scripts/config/env.js

```diff
@@ -24,8 +24,12 @@ if (!NODE_ENV) {
   );
 }
 
+const DOTENV = process.env.DOTENV;
```

Comment:
Please change the name of this variable to `REACT_APP_ENV`.

We will also need tests for this functionality in `packages/react-scripts/fixtures/kitchensink/integration/env.test.js`